src/app/clusters/unit-localization-server/unit-localization-server.cpp

```diff
@@ -132,7 +134,9 @@ CHIP_ERROR UnitLocalizationServer::SetTemperatureUnit(TempUnitEnum newTempUnit)
         }
     }
     VerifyOrReturnError(isValid, CHIP_IM_GLOBAL_STATUS(ConstraintError));
+    VerifyOrReturnValue(mTemperatureUnit != newTempUnit, CHIP_NO_ERROR);
     mTemperatureUnit = newTempUnit;
+    MatterReportingAttributeChangeCallback(kRootEndpointId, UnitLocalization::Id, TemperatureUnit::Id);
```

Comment:
This likely should be tested by the cert test.

Reply:
This is fairly a simple condition but if the work for test plan happens, I will gladly put the test implementation.